core/src/main/scala/kafka/controller/KafkaController.scala

```
@@ -1223,19 +1195,21 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     }
 
     try {
-      zkClient.registerController(config.brokerId, timestamp)
-      info(s"${config.brokerId} successfully elected as the controller")
+      tryRegisterController(timestamp)
+      info(s"${config.brokerId} successfully elected as the controller. Epoch incremented to ${controllerContext.epoch} " +
+        s"and epoch zk version is now ${controllerContext.epochZkVersion}")
+
       activeControllerId = config.brokerId
       onControllerFailover()
     } catch {
-      case _: NodeExistsException =>
+      case e1: ControllerMovedException =>
```

If `onControllerFailover()` throws `ControllerMovedException` after controller has registered itself as controller, it seems possible that some events may have already been inserted into the controller event queue. Should we propagate the `ControllerMovedException` to `ControllerEventThread` in order to clear the controller event queue?

Also, I think in most places we will just name the exception varaible as `e` and throwable variable as `t`. Naming them e1, e2, and specifically naming a throwalble as `e2`, seems unusual. I know this style is used in the existing controller code. I am wondering if we can change this.

---

You are right. We should propagate the exception here.
Done.